### PR TITLE
Bump phpstan to level 6

### PR DESCRIPTION
**Pull request by VincentLanglet**
Need
- https://github.com/sonata-project/SonataAdminBundle/pull/6967
- https://github.com/doctrine/orm/pull/8556
- https://github.com/phpstan/phpstan-src/pull/476
- https://github.com/sonata-project/EntityAuditBundle/pull/375

### CODE REVIEW

File: tests/App/Admin/CategoryAdmin.php [hunk 1]
```diff
@@ -20,6 +20,9 @@
 use Sonata\AdminBundle\Show\ShowMapper;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 
+/**
+ * @phpstan-extends AbstractAdmin<\Sonata\DoctrineORMAdminBundle\Tests\App\Entity\Category>
```
Comment (jordisala1991) [inline tests/App/Admin/CategoryAdmin.php:24]: isn't it beter to use a use?

Comment (VincentLanglet) [reply tests/App/Admin/CategoryAdmin.php:24]: IDE like phpstorm are not understanding that the class will be used.
It could be confusing.

**Comment by VincentLanglet** [comment]
This should be ready @sonata-project/contributors :)

**Review by jordisala1991 — DISMISSED** [review]

File: tests/Builder/ShowBuilderTest.php [hunk 2]
```diff
@@ -28,8 +30,19 @@
...
+     */
     private $guesser;
+
+    /**
+     * @var ShowBuilder
+     */
     private $showBuilder;
+
+    /**
+     * @var MockObject|AdminInterface<object>
```
Comment (franmomu) [inline tests/Builder/ShowBuilderTest.php:44]:
```suggestion
     * @var MockObject&AdminInterface<object>
```

File: tests/Builder/ShowBuilderTest.php [hunk 3]
```diff
@@ -28,8 +30,19 @@
  */
 class ShowBuilderTest extends TestCase
 {
+    /**
+     * @var Stub|TypeGuesserInterface
```
Comment (franmomu) [inline tests/Builder/ShowBuilderTest.php:34]:
```suggestion
     * @var Stub&TypeGuesserInterface
```

File: tests/Datagrid/PagerTest.php [hunk 4]
```diff
@@ -32,6 +35,8 @@ public function entityClassDataProvider(): array
     }
 
     /**
+     * @param class-string $className
```
Comment (franmomu) [inline tests/Datagrid/PagerTest.php:38]:
```suggestion
     * @phpstan-param class-string $className
```
I guess `class-string` is not regular phpdoc, if not, there are others

File: tests/Datagrid/ProxyQueryTest.php [hunk 5]
```diff
@@ -60,6 +60,7 @@ protected function setUp(): void
 
     protected function tearDown(): void
     {
+        // @phpstan-ignore-next-line
```
Comment (franmomu) [inline tests/Datagrid/ProxyQueryTest.php:63]: why is this one needed?

File: tests/Fixtures/DoctrineType/ValueObjectWithToStringImpl.php [hunk 6]
```diff
@@ -15,20 +15,23 @@
 
 final class ValueObjectWithToStringImpl
```
Comment (franmomu) [inline tests/Fixtures/DoctrineType/ValueObjectWithToStringImpl.php:16]: now looking at this one, is similar to the previous one, I don't think `$uuid` was meant to be a string.

File: tests/Model/ModelManagerTest.php [hunk 7]
```diff
@@ -82,7 +82,10 @@ protected function setup(): void
         $this->modelManager = new ModelManager($this->registry, PropertyAccess::createPropertyAccessor());
     }
 
-    public function valueObjectDataProvider(): array
+    /**
+     * @return iterable<array{class-string}>
```
Comment (franmomu) [inline tests/Model/ModelManagerTest.php:86]:
```suggestion
     * @phpstan-return iterable<array{class-string}>
```

File: tests/Datagrid/ProxyQueryTest.php (hunk 5 above)
Comment (VincentLanglet) [reply tests/Datagrid/ProxyQueryTest.php:63]: Cause we're setting null to a variable which is not nullable.

Maybe a PR could be made to phpstan-phpunit

File: tests/Fixtures/DoctrineType/ValueObjectWithMagicToStringImpl.php [hunk 8]
```diff
@@ -15,20 +15,23 @@
...
     {
-        return (string) $this->uuid;
+        return $this->uuid;
     }
 
-    public function getValue()
+    public function getValue(): string
     {
-        return (string) $this;
+        return $this->uuid;
```
[the remark this answers is not in the document]
Comment (VincentLanglet) [reply tests/Fixtures/DoctrineType/ValueObjectWithMagicToStringImpl.php:35]: The usage is the following one 
```
new $vbClassName('a7ef873a-e7b5-11e9-81b4-2a2ae2dbcce4')
```

The only  things important here is that this class implement `__toString` and the other `toString`.

File: tests/Datagrid/ProxyQueryTest.php (hunk 5 above)
Comment (VincentLanglet) [reply tests/Datagrid/ProxyQueryTest.php]: I think I have a better solution.
Using `unset($this->em);`

File: tests/Fixtures/Entity/DoubleNameEntity.php [hunk 9]
```diff
@@ -20,15 +20,29 @@
 /** @Entity */
 final class DoubleNameEntity
 {
-    /** @Column(type="string") */
+    /**
+     * @Column(type="string")
+     *
+     * @var string|null
```
Comment (franmomu) [inline tests/Fixtures/Entity/DoubleNameEntity.php:26]:
```suggestion
     * @var string
```

File: tests/Fixtures/Entity/DoubleNameEntity.php [hunk 10]
```diff
@@ -20,15 +20,29 @@
...
+     */
     public $name2;
-    /** @Id @Column(type="integer") */
+
+    /**
+     * @Id
+     * @Column(type="integer")
+     *
+     * @var int|null
```
Comment (franmomu) [inline tests/Fixtures/Entity/DoubleNameEntity.php:41]:
```suggestion
     * @var int
```

File: tests/Fixtures/Entity/UuidBinaryEntity.php [hunk 11]
```diff
@@ -15,13 +15,22 @@
 
 final class UuidBinaryEntity
 {
+    /**
+     * @var string|object
```
Comment (franmomu) [inline tests/Fixtures/Entity/UuidBinaryEntity.php:19]:
```suggestion
     * @var object
```
just object?

**Comment by VincentLanglet** [comment]
Done @franmomu :)

**Review by franmomu — APPROVED** [review]

**Pull request merged**